core/src/test/scala/matryoshka/patterns/diff.scala

```diff
@@ -0,0 +1,107 @@
+class DiffSpec extends Specification with DiffArb with specs2.scalaz.Matchers {
+  "diff" should {
+    "find non-recursive differences" in {
+      NonRec[Mu[Example]]("a", 1).embed.paraMerga(NonRec[Mu[Example]]("b", 1).embed)(diff) must
+      equal(CorecursiveOps[Mu, Diff[Mu, Example, ?]](
+        LocallyDifferent(NonRec[DiffT[Mu, Example]]("a", 1),
+                         NonRec[Unit]("b", 1))).embed)
+    }
+
+    "create `Removed` for shorter list" in {
+      OneList(List(Empty[Mu[Example]]().embed, Empty[Mu[Example]]().embed)).embed.paraMerga(
+        OneList(List(NonRec[Mu[Example]]("x", 3).embed)).embed)(diff) must
+        equal(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Similar(OneList(List(
+          CorecursiveOps[Mu, Diff[Mu, Example, ?]](Different(Empty[Mu[Example]]().embed, NonRec[Mu[Example]]("x", 3).embed)).embed,
+          CorecursiveOps[Mu, Diff[Mu, Example, ?]](Removed(Empty[Mu[Example]]().embed)).embed)))).embed)
+    }
+
+    "create `Added` for longer list" in {
+      OneList(List(NonRec[Mu[Example]]("x", 3).embed)).embed.paraMerga(
+        OneList(List(Empty[Mu[Example]]().embed, Empty[Mu[Example]]().embed)).embed)(diff) must
+        equal(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Similar(OneList(List(
+          CorecursiveOps[Mu, Diff[Mu, Example, ?]](Different(NonRec[Mu[Example]]("x", 3).embed, Empty[Mu[Example]]().embed)).embed,
+          CorecursiveOps[Mu, Diff[Mu, Example, ?]](Added(Empty[Mu[Example]]().embed)).embed)))).embed)
+    }
+
+    "choose “simplest” diff" in {
+      OneList(List(Empty[Mu[Example]]().embed)).embed.paraMerga(
+        OneList(List(NonRec[Mu[Example]]("x", 3).embed, Empty[Mu[Example]]().embed)).embed)(diff) must
+        equal(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Similar(OneList(List(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Added(NonRec[Mu[Example]]("x", 3).embed)).embed, CorecursiveOps[Mu, Diff[Mu, Example, ?]](Same(Empty[Mu[Example]]().embed)).embed)))).embed)
+    }.pendingUntilFixed("can’t just walk lists from start to finish")
+
+    "create `Removed` and `Added` for mixed lists" in {
+      TwoLists(List(Empty[Mu[Example]]().embed, Empty[Mu[Example]]().embed), List(Empty[Mu[Example]]().embed)).embed.paraMerga(
+        TwoLists(
+          List(NonRec[Mu[Example]]("x", 3).embed),
+          List(Empty[Mu[Example]]().embed, Empty[Mu[Example]]().embed, Empty[Mu[Example]]().embed)).embed)(diff) must
+        equal(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Similar(TwoLists(
+          List(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Different(Empty[Mu[Example]]().embed, NonRec[Mu[Example]]("x", 3).embed)).embed, CorecursiveOps[Mu, Diff[Mu, Example, ?]](Removed(Empty[Mu[Example]]().embed)).embed),
+          List(CorecursiveOps[Mu, Diff[Mu, Example, ?]](Same(Empty[Mu[Example]]().embed)).embed, CorecursiveOps[Mu, Diff[Mu, Example, ?]](Added(Empty[Mu[Example]]().embed)).embed, CorecursiveOps[Mu, Diff[Mu, Example, ?]](Added(Empty[Mu[Example]]().embed)).embed)))).embed)
+    }
+  }
+}
```

Comment:
These tests don't really convince me that this Diff thing is useful (except as a tricky example). Maybe add a test of the `Show` instance, since it might read in a recognizable way?